src/django_components/attributes.py

```diff
@@ -2,39 +2,84 @@
 # See https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/templatetags/components.py  # noqa: E501
 # And https://github.com/Xzya/django-web-components/blob/b43eb0c832837db939a6f8c1980334b0adfdd6e4/django_web_components/attributes.py  # noqa: E501
 
-from typing import Any, Dict, List, Mapping, Optional, Tuple
+from typing import Any, Dict, Mapping, Optional, Tuple
 
 from django.template import Context
 from django.utils.html import conditional_escape, format_html
 from django.utils.safestring import SafeString, mark_safe
 
 from django_components.node import BaseNode
-from django_components.util.template_tag import TagParams
-
-HTML_ATTRS_DEFAULTS_KEY = "defaults"
-HTML_ATTRS_ATTRS_KEY = "attrs"
 
 
 class HtmlAttrsNode(BaseNode):
```

Comment:
Now, all these 4 are co-located:
1. Tag definition (e.g. whether it's `{% slot %}` or `{% provide %}`),
2. The Node class that the tag is converted to
3. The `render` logic when said Node is encountered during render
4. The source of truth for the API documentation of this template tag

So I refactored all our template tags, from `templatetags/component_tags.py`, so that the actual definitions are stored in the respective files. So now, `templatetags/component_tags.py` contains no logic, and it only declares which template tags we define.

Previously the documentation for individual tags was in `templatetags/component_tags.py`, so that's why you can see the docstrings here. The docstrings didn't change.